### PR TITLE
OpenTelemetry: initial metrics and export via otlpmetrichttp

### PR DESCRIPTION
**Pull request by heavycrystal**
`flow-worker` now exports `slotlag` and `open_connections` metrics (both are gauges) for each Postgres peer. Metrics enabled by the environment variable `ENABLE_OTEL_METRICS` and export endpoint set by the environment variable `OTEL_EXPORTER_OTLP_METRICS_ENDPOINT` on `flow-worker`.

Metrics update at the frequency of the `RecordSlotSizesWorkflow`, which is currently 5 minutes.

### CODE REVIEW

File: flow/shared/otel_metrics.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+func (g *Int64Gauge) Set(val int64) {
+	g.currentVal.Store(val)
+}
+
+type Float64Gauge struct {
+	observableGauge metric.Float64ObservableGauge
+	floatMutex      sync.Mutex
+	currentVal      float64
```
Comment (serprex) [inline flow/shared/otel_metrics.go:42]: Use atomic.Int64 & math.Float64frombits

File: flow/activities/flowable.go [hunk 2]
```diff
@@ -592,7 +603,43 @@ func (a *FlowableActivity) RecordSlotSizes(ctx context.Context) error {
 			if ctx.Err() != nil {
 				return
 			}
-			err = srcConn.HandleSlotInfo(ctx, a.Alerter, a.CatalogPool, slotName, peerName)
+
+			var slotLagGauge *shared.Float64Gauge
```
Comment (iskakaushik) [inline flow/activities/flowable.go:607]: can you refactor this to a method?

**Review by serprex — APPROVED** [review]

File: flow/connectors/postgres/postgres.go [hunk 3]
```diff
@@ -1133,6 +1141,9 @@ func (c *PostgresConnector) HandleSlotInfo(
 		return err
 	}
 	alerter.AlertIfOpenConnections(ctx, peerName, res)
+	openConnectionsGauge.Set(res.CurrentOpenConnections, attribute.NewSet(
+		attribute.String("peerName", peerName),
+		attribute.String("deploymentUID", peerdbenv.PeerDBDeploymentUID())))
```
[the remark this answers is not in the document]
Comment (heavycrystal) [reply flow/connectors/postgres/postgres.go:1147]: nil checks present in `.Set()`

File: flow/otel_metrics/otel_manager.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+		resource.NewWithAttributes(
+			semconv.SchemaURL,
+			semconv.ServiceNameKey.String(otelServiceName),
+		),
+	)
+
+	return r, err
+}
+
+func SetupOtelMetricsExporter() (*sdkmetric.MeterProvider, error) {
```
Comment (iamKunalGupta) [inline flow/otel_metrics/otel_manager.go:35]: can modify to accept the service name (`flow-worker`) to make it service agnostic (and reusable across `api`, `worker`, `snapshot`).

**Review by iamKunalGupta — APPROVED** [review]

File: flow/otel_metrics/otel_manager.go (hunk 4 above)
Comment (heavycrystal) [reply flow/otel_metrics/otel_manager.go]: changed

**Pull request merged**